test/common/routes_exist.go

```diff
@@ -0,0 +1,209 @@
+package common
+
+import (
+	"fmt"
+	"testing"
+
+	goctx "context"
+
+	routev1 "github.com/openshift/api/route/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var expectedRoutes = map[string][]ExpectedRoute{
+	"3scale": []ExpectedRoute{
+		ExpectedRoute{
+			Name:  "backend",
+			isTLS: true,
+		},
+		ExpectedRoute{
+			Name:            "zync-3scale-api-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "apicast-staging",
+		},
+		ExpectedRoute{
+			Name:            "zync-3scale-api-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "apicast-production",
+		},
+		ExpectedRoute{
+			Name:            "zync-3scale-master-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "system-master",
+		},
+		ExpectedRoute{
+			Name:            "zync-3scale-provider-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "system-developer",
+		},
+		ExpectedRoute{
+			Name:            "zync-3scale-provider-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "system-provider",
+		},
+	},
+
+	"amq-online": []ExpectedRoute{
+		ExpectedRoute{
+			Name:  "console",
+			isTLS: true,
+		},
+		ExpectedRoute{
+			Name:  "standard-authservice",
+			isTLS: true,
+		},
+	},
+
+	"codeready-workspaces": []ExpectedRoute{
+		ExpectedRoute{
+			Name:  "codeready",
+			isTLS: true,
+		},
+		ExpectedRoute{
+			Name:  "devfile-registry",
+			isTLS: true,
+		},
+		ExpectedRoute{
+			Name:  "plugin-registry",
+			isTLS: true,
+		},
+	},
+
+	"middleware-monitoring-operator": []ExpectedRoute{
+		ExpectedRoute{
+			Name:  "alertmanager-route",
+			isTLS: true,
+		},
+		ExpectedRoute{
+			Name:  "grafana-route",
+			isTLS: true,
+		},
+		ExpectedRoute{
+			Name:  "prometheus-route",
+			isTLS: true,
+		},
+	},
+
+	"rhsso": []ExpectedRoute{
+		ExpectedRoute{
+			Name:  "keycloak",
+			isTLS: false,
+		},
+		ExpectedRoute{
+			Name:  "keycloak-edge",
+			isTLS: true,
+		},
+	},
+
+	"solution-explorer": []ExpectedRoute{
+		ExpectedRoute{
+			Name:  "tutorial-web-app",
+			isTLS: true,
+		},
+	},
+
+	"ups": []ExpectedRoute{
+		ExpectedRoute{
+			Name:  "ups-unifiedpush-proxy",
+			isTLS: true,
+		},
+	},
+
+	"user-sso": []ExpectedRoute{
+		ExpectedRoute{
+			Name:  "keycloak",
+			isTLS: false,
+		},
+		ExpectedRoute{
+			Name:  "keycloak-edge",
+			isTLS: true,
+		},
+	},
+}
+
+// TestIntegreatlyRoutesExist tests that the routes for all the products are created
+func TestIntegreatlyRoutesExist(t *testing.T, ctx *TestingContext) {
+	for product, routes := range expectedRoutes {
+		for _, expectedRoute := range routes {
+			foundRoute, err := getRoute(t, ctx, product, expectedRoute)
+
```

Comment:
```suggestion
			foundRoute, err := getRoute(t, ctx, product, expectedRoute)
```

Reply:
Just the new line :sweat_smile: , usually the if err is attached to the function call

Reply:
Oops, the diff didn't show it. I just pushed that change!